src/auth/connect.ts

```diff
@@ -0,0 +1,82 @@
+import { OnePasswordConnect } from "@1password/connect";
+import { OPConnect } from "@1password/connect/dist/lib/op-connect";
+import { parseSecretRef } from "../utils";
+import type { SecretReferenceResolver } from "./types";
+
+export class Connect implements SecretReferenceResolver {
+	private op: OPConnect;
+
+	public constructor(serverURL: string, token: string) {
+		this.op = OnePasswordConnect({
+			serverURL,
+			token,
+			keepAlive: true,
+		});
+	}
+
+	public async resolve(ref: string): Promise<string> {
+		const secret = await this.resolveByPath(ref);
+		if (!secret) {
+			throw new Error(`Can't resolve this ${ref}`);
+		}
+		return secret;
+	}
+
+	/**
+	 * This is method derive from https://github.com/1Password/onepassword-operator/blob/ced45c33d4c1e0267dc5af54231c5a29accce4c4/pkg/onepassword/items.go
+	 * @param path Secret reference path, match `ref_regex`
+	 */
+	private async resolveByPath(path: string): Promise<string | undefined> {
+		const matched = parseSecretRef(path);
+		if (!matched) {
+			throw new Error(`Invalid secret reference: ${path}`);
+		}
+		const { vaultName, itemName, sectionName, fieldName } = matched;
+
+		const vault = await this.op.getVault(vaultName);
+		if (!vault.id) {
+			return undefined;
+		}
+		const item = await this.op.getItem(vault.id, itemName);
+
+		let itemFields = item.fields;
+		const errFiledName = sectionName
+			? `${sectionName}.${fieldName}`
+			: fieldName;
+		if (sectionName) {
+			const section = item.sections?.filter(
+				(s) => s.label === sectionName || s.id === sectionName,
+			);
+			if (section === undefined || section.length === 0) {
+				throw new Error(`The item does not have a field '${errFiledName}'`);
+			}
+			if (section.length > 1) {
+				throw new Error("More than one section matched the secret reference");
+			}
+			const sectionId = section[0]!.id;
+			itemFields = itemFields?.filter((f) => f.section?.id === sectionId);
+		}
+
+		const matchedFields = itemFields?.filter(
+			(f) => f.id === fieldName || f.label === fieldName,
+		);
+
+		if (matchedFields === undefined || matchedFields.length === 0) {
+			throw new Error(`The item does not have a field '${errFiledName}'`);
+		}
+		if (matchedFields.length > 1) {
+			// if section not provide, return the default section id equals to "add more" filed
+			if (!sectionName) {
+				const res = matchedFields.find(
+					(f) => f.section?.id === "add more",
+				)?.value;
+				if (res) {
+					return res;
+				}
+			}
```

Comment:
For Connect the following reference will fail: `op://acceptance-tests/test-secret/password`.

The `password` field isn't in a section with the ID `add-more`, but rather the section ID is empty. The default section in an item is either with the ID `add-more` or the empty string.

Reply:
@edif2008 all done, PTAL